src/core/internal/lifetime.d

```diff
@@ -89,37 +89,39 @@ Emplaces T.init.
 In contrast to `emplaceRef(chunk)`, there are no checks for disabled default
 constructors etc.
 +/
-void emplaceInitializer(T)(scope ref T chunk) nothrow pure @trusted
-    if (!is(T == const) && !is(T == immutable) && !is(T == inout))
+template emplaceInitializer(T)
+if (!is(T == const) && !is(T == immutable) && !is(T == inout))
 {
     import core.internal.traits : hasElaborateAssign;
 
-    static if (!hasElaborateAssign!T && __traits(compiles, chunk = T.init))
-    {
-        chunk = T.init;
-    }
-    else static if (__traits(isZeroInit, T))
-    {
-        static if (is(T U == shared U))
-            alias Unshared = U;
-        else
-            alias Unshared = T;
+    // Avoid stack allocation by hacking to get to the init symbol.
+    pragma(mangle, "_D" ~ T.mangleof[1..$] ~ "6__initZ")
+    __gshared extern immutable typeof(T.init) initializer;
```

Comment:
This is only used in the `else` block

Reply:
Indeed. 
I guess your comment is to static-if this away, but I would have to replicate the full static-if chain, and it does not harm to have it here.

Reply:
You could move it into the `else` block?

Reply:
it needs to be a global variable  (and what i remember is that making it static did not work)

Reply:
It's just a fwd decl, so no harm done if unused.

Reply:
That's weird but should not block this PR.

Reply:
> It's just a fwd decl, so no harm done if unused.

Not for LDC:
```
FAILED: runtime/objects-unittest-debug/std/experimental/allocator/building_blocks/affix_allocator.o 
cd /home/runner/work/ldc/ldc/runtime/phobos && /home/runner/work/ldc/ldc/bin/ldc2 -c --output-o -conf= -w -de -dip1000 -preview=dtorfields -g -link-defaultlib-debug -d-debug -d-version=StdUnittest -unittest -linkonce-templates -relocation-model=pic -I/home/runner/work/ldc/ldc/runtime/druntime/src -I/home/runner/work/ldc/ldc/runtime/phobos -of=/home/runner/work/ldc/ldc/runtime/objects-unittest-debug/std/experimental/allocator/building_blocks/affix_allocator.o std/experimental/allocator/building_blocks/affix_allocator.d
/home/runner/work/ldc/ldc/runtime/druntime/src/core/internal/lifetime.d(99): Error: Global variable type does not match previous declaration with same mangled name: `_D6__initZ`
/home/runner/work/ldc/ldc/runtime/druntime/src/core/internal/lifetime.d(99):        Previous IR type: i64, const, non-thread-local
/home/runner/work/ldc/ldc/runtime/druntime/src/core/internal/lifetime.d(99):        New IR type:      i32, const, non-thread-local
```